keps/0001-universal-operator.md

```diff
@@ -0,0 +1,18 @@
+# Maestro: A Universal Operator
+
+## Summary
+
+We propose a universal operator that generates and runs operators based on a standard data format. This universal operator manages the lifecycle of stateful workloads.
+
+## Motivation
+
+While Kubebuilder provides an abstraction around the plumbing for building operators, there is still a lot of work to implement operators in code. To combat this problem, Mesosphere built an SDK for DC/OS, as well as a package data format, that handles a lot of the common high level tasks in running stateful workloads and provides lifecycle hooks. The purpose of Maestro is to take this work and enable the same standard packaging format and lifecycle hooks to be applied to Kubernetes operators.
+
```

Comment:
It will be good to describe what the standard packaging format looks like. It doesn't have to be detailed, but good enough to give an idea what does input look like.

Reply:
> It will be good to describe what the standard packaging format looks like. It doesn't have to be detailed, but good enough to give an idea what does input look like.

Agree.  I'm particularly interested in a rough sketch of a versioning story in a multi-tenant cluster.